### PR TITLE
Transform pasted div tags to p tags

### PR DESCRIPTION
**Pull request by adamziel**
## Description
Solves https://github.com/WordPress/gutenberg/issues/34160 

With this PR applied, content pasted from Evernote is no longer collapsed into the same paragraph:

## How has this been tested?
1. Type a few paragraphs of text in Evernote
2. Copy and paste it to Gutenberg
3. Confirm the formatting was preserved

## Types of changes
Bug fix (non-breaking change which fixes an issue)

cc @ellatrix @getdave @jorgefilipecosta @draganescu @noisysocks

### CODE REVIEW

File: packages/blocks/src/api/raw-handling/test/phrasing-content-reducer.js [hunk 1]
```diff
@@ -51,6 +51,12 @@ describe( 'phrasingContentReducer', () => {
 		);
 	} );
 
+	it( 'should transform divs to paragraphs', () => {
+		expect(
+			deepFilterHTML( '<div>test</div>', [ phrasingContentReducer ], {} )
```
Comment (getdave) [inline packages/blocks/src/api/raw-handling/test/phrasing-content-reducer.js]: What happens with this testing content?

```
'<div>test<div>this is test</div></div>',
```

Are we going to end up with situations whereby nested divs turn into nested `<p>` tags? I'm not sure nesting is valid.

**Review by getdave — COMMENTED** [review]
Thanks for the PR!

This seems like a nice addition but I"m concerned about generating invalid markup. Am I right in saying it wouldn't be great to end up with nested `<p>` tags?

File: packages/blocks/src/api/raw-handling/test/phrasing-content-reducer.js (hunk 1 above)
Comment (adamziel) [reply packages/blocks/src/api/raw-handling/test/phrasing-content-reducer.js]: @getdave good point! It does nest the p tags unfortunately. One of the other filters separates them, but I don't like that we end up in a broken state in the first place.

Also, I just reviewed the [phrasing content](https://html.spec.whatwg.org/dev/dom.html#phrasing-content-2) part of the DOM specs, and div tags are not a phrasing content. Perhaps we need to introduce a new filter here and use some smart ordering.

**Comment by adamziel** [comment]
@getdave I revisited the approach used in this PR. What do you think now?

**Review by getdave — APPROVED** [review]
I like the way it removes the wrapping div and then normalizes everything else.

I dont have a tonne of knowledge about potential knock ons here but no tests broke and this one now handles the nesting use case so I'm hopeful this will be ok.

Nice work!

**Pull request merged**